content/usgs-packages/Application_Part3.Rmd

```diff
@@ -0,0 +1,182 @@
+
+set.seed(1)
+```
+
+```{r sbtools-auth, echo=FALSE}
+# run vizlab::storeSBcreds() once before this can work
+credList <- readRDS(file.path(path.expand('~'), ".vizlab/sbCreds"))
+un <- rawToChar(credList$username)
+pw <- rawToChar(credList$password)
+sbtools::authenticate_sb(un, pw)
+```
+
+In this section, you will use data downloaded from the [previous section](/usgs-packages/app-part2) to create time series plots and a map. The goal is to have 3 timeseries plots per site/HUC (precipitation, nitrogen concentration, and phosphorus concentration), and one summary map of cumulative precipition for each HUC.
+
+This will be put together as a single script at the end, but use the code below to remind yourself what we have done up to this point.
+
+Getting sites and data:
+
+```{r get-sites-data, addToggle="ON", message=FALSE, warning=FALSE}
+library(sbtools)
+library(dataRetrieval)
+library(geoknife)
+
+# identify site id and query for files
+sb_site_id <- "59848b35e4b0e2f5d46717d1"
+avail_files <- item_list_files(sb_site_id)
+
+# use appropriate reader to get file (tab delimited) into R & get site numbers
+sb_sites_df <- read.table(avail_files$url[1], sep="\t", header=TRUE,
+                          colClasses = "character", stringsAsFactors = FALSE)
+sites <- sb_sites_df$site_number
+
+# get HUC 8 codes for precip data
+sb_sites_info <- readNWISsite(sites)
+huc8s <- sb_sites_info$huc_cd
+
+# define period
+startDate <- "2015-10-01"
+endDate <- "2016-09-30"
+
+# download nutrient data
+pcodes_nitrogen <- c("00613", "00618", "00631")
+pcodes_phosphorus <- c("00665")
+nitrogen_data <- readNWISqw(siteNumbers = sites, parameterCd = pcodes_nitrogen,
+                            startDate = startDate, endDate = endDate)
+phosphorus_data <- readNWISqw(siteNumbers = sites, parameterCd = pcodes_phosphorus,
+                              startDate = startDate, endDate = endDate)
+
+# download precip data
+precip_stencil <- webgeom(paste0('HUC8::', paste(huc8s, collapse=",")))
+precip_knife <- webprocess() # accept defaults for weighted average
+all_webdata <- query("webdata")
+precip_fabric <- webdata(all_webdata["United States Stage IV Quantitative Precipitation Archive"])
+variables(precip_fabric) <- query(precip_fabric, 'variables')
+times(precip_fabric) <- c(startDate, endDate)
+```
+
+Executing the geojob:
+
+```{r execute-job-off, addToggle="ON", eval=FALSE}
+precip_geojob <- geoknife(precip_stencil, precip_fabric, precip_knife)
+wait(precip_geojob, sleep.time = 10) # add `wait` when running scripts
+precip_data <- result(precip_geojob)
+```
+
+```{r read-intermediate, echo=FALSE}
+geoknife_list <- readRDS("precip_geoknife.RDS")
+precip_geojob <- geoknife_list$geojob
+precip_data <- geoknife_list$data
+```
+
+## Prepare data for plotting
+
+First re-organize the precipitation into a long format (one column for value, one for HUC) rather than a separate column of precip for each HUC. See `gather` from the package `tidyr` for ideas on how to do it.
```

Comment:
"long format (one column for value, one for HUC) rather than a separate column of precip for each HUC." -> "long format: one column for HUC and one for precip, rather than a separate column of precip for each HUC."